src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -0,0 +1,197 @@
+            throws ServletException, IOException {
+        proxyWmsRequest(request, response);
+    }
+
+    private void proxyWmsRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        proxyWmsRequest(request, response, "LAYERS");
+    }
+
+    private void proxyWmsRequest(HttpServletRequest request, HttpServletResponse response, String layerParameter)
+            throws ServletException, IOException {
+        LayerDescriptor layerDescriptor = new LayerDescriptor(request.getParameter(layerParameter)); // TODO needs to be case insensitive
+
+        String time = request.getParameter("TIME");
+
+        String wmsUrlStr = getWmsUrl(layerDescriptor, time);
+
+        try {
+            Map<String, String[]> wmsParameters = new HashMap(request.getParameterMap());
+            wmsParameters.remove("TIME");
+            wmsParameters.put("VERSION", new String[] { wmsVersion });
+            wmsParameters.put(layerParameter, new String[] { layerDescriptor.variable });
+
+            // Needed for GetFeatureInfo
+            if (wmsParameters.containsKey("QUERY_LAYERS")) {
+                wmsParameters.put("QUERY_LAYERS", new String[] { layerDescriptor.variable });
+            }
+
+
+            String queryString = encodeMapForRequest(wmsParameters);
+
+            URL wmsUrl = new URL(wmsUrlStr + "?" + queryString);
+
+            IOUtils.copy(wmsUrl.openConnection().getInputStream(), response.getOutputStream());
+        }
+        catch (Exception e) {
+            LOGGER.log(Level.SEVERE, String.format("Problem proxying url '%s'", wmsUrlStr));
+            e.printStackTrace();
+        }
+    }
+
+    private String encodeMapForRequest(Map<String, String[]> params) {
+        StringBuilder sb = new StringBuilder();
+        try {
+            for (Map.Entry<String, String[]> param : params.entrySet()) {
+                if (sb.length() != 0) sb.append('&');
+                sb.append(URLEncoder.encode(param.getKey(), StandardCharsets.UTF_8.name()));
+                sb.append('=');
+                sb.append(URLEncoder.encode(param.getValue()[0], StandardCharsets.UTF_8.name()));
+            }
+        }
+        catch (Exception e) {
+            LOGGER.log(Level.SEVERE, String.format("Error encoding parameters: '%s'", e.getMessage()));
+        }
+
+        return sb.toString();
+    }
+
+    private static List<String> getCombinedStyles(Document getCapabilitiesXml, String layerName) {
+        List<String> combinedStyles = new ArrayList<String>();
+
+        DefaultXPath xpath = new DefaultXPath("//x:Layer/x:Title[.=\'" + layerName + "\']/../x:Style/x:Name/text()");
+        Map<String,String> namespaces = new TreeMap<String, String>();
+        namespaces.put("x", "http://www.opengis.net/wms");
+        xpath.setNamespaceURIs(namespaces);
+        List<DefaultText> list = xpath.selectNodes(getCapabilitiesXml);
```

Comment:
Not quite sure I follow here. `xpath.selectNodes(xml)` returns `java.util.List<org.jdom4.tree.DefaultText>`. So why is it not matching?